driver-core/src/main/java/com/datastax/driver/core/PagingState.java

```diff
@@ -0,0 +1,176 @@
+            return false;
+        }
+        ByteBuffer hashToCheck = extractHashDigest(this.pagingState);
+        ByteBuffer extractedPagingState = extractPagingState(this.pagingState);
+        ByteBuffer toTest = this.hash(statement, extractedPagingState);
+        return hashToCheck.equals(toTest);
+    }
+
+    private ByteBuffer extractPagingState(ByteBuffer complete) {
+        complete.rewind();
+        short size = complete.getShort();
+        byte[] array = new byte[size];
+        pagingState.position(pagingState.position() + 2);
+        complete.get(array);
+        return ByteBuffer.wrap(array);
+    }
+
+    private ByteBuffer extractHashDigest(ByteBuffer complete) {
+        complete.rewind();
+        short pagingSize = complete.getShort();
+        short hashSize = complete.getShort();
+        complete.position(pagingSize + 4);
+        byte[] array = new byte[hashSize];
+        complete.get(array);
+        return ByteBuffer.wrap(array);
+    }
+
+    private ByteBuffer generatePagingState(ByteBuffer pagingState, ByteBuffer hash) {
+        short pagingSize = (short)pagingState.remaining();
+        short hashSize = (short)hash.remaining();
+        ByteBuffer res = ByteBuffer.allocate(pagingSize + hashSize + 4);
+
+        res.putShort(pagingSize);
+        res.putShort(hashSize);
+
+        res.put(pagingState);
+        res.put(hash);
+
+        res.flip();
+        return res;
+    }
+
+    ByteBuffer getRawState() {
+        return extractPagingState(this.pagingState);
+    }
+
+    @Override
+    public String toString() {
+        return Bytes.toRawHexString(this.pagingState);
+    }
+
+    /**
+     * Create a PagingState object from a string previously generated with {@link #toString()}.
+     *
+     * @param string the string value.
+     * @return the PagingState object created.
+     */
+    public static PagingState fromString(String string) {
+        try {
+            ByteBuffer complete = ByteBuffer.wrap(Bytes.fromRawHexString(string, 0));
+            return new PagingState(complete);
+        } catch (Exception e) {
+            throw new PagingStateException("String format doesn't match for generating the paging state object. "
+                + "MUST be a string from a previously saved PagingState object. Nested exception : " + e);
+        }
+    }
+
+    /**
+     * Return a representation of the paging state object as a bytes array.
+     *
+     * @return the paging state as a bytes array.
+     */
+    public byte[] toBytes() {
+        return this.pagingState.array();
+    }
+
+    /**
+     * Create a PagingState object from a byte array previously generated with {@link #toBytes()}.
+     *
+     * @param pagingState The bytes array representation.
+     * @return the PagingState object created.
+     */
+    public static PagingState fromBytes(byte[] pagingState) {
+        try {
+            return new PagingState(ByteBuffer.wrap(pagingState));
+        } catch (Exception e) {
```

Comment:
Alright for the exception, but for the verification, since it is automatically made when we set the paging state I don't think it would be useful to do it there again. In addition to that, the hash is composed of a statement query and params so we cannot generate one here since no statement is concerned